doc/tests/verification_plan.rst

```diff
@@ -0,0 +1,138 @@
+    driver functionalities are verified.
+
+**Power management services**
+    Power management services such as tickless idle and an advanced idling
+    infrastructure are also verified.
+
+**POSIX compliance**
+    APIs introduced to make Zephyr POSIX complaint are verified through API level testing.
+
+Verification Strategy and Approach
+##################################
+
+Zephyr test suite contains various test types such as
+
+#. Component level test
+
+   Test against single Zephyr module.
+
+#. API level test
+
+   Test correctness of Zephyr APIs made available for application developer.
+
+#. Stress test
+
+   Thorough and intense tests to ensure robustness of Zephyr components.
+
+#. Security test
+
+   Intended to reveal flaws in the security mechanisms such memory protection
+   implementation and stack corruption check.
+
+#. Performance test
+
+   Testing done against key performance indicators to ensure effectiveness
+   of Zephyr.
+
+#. Integration test
+
+   Individual components are combined and verified as a group.
+
+Zephyr Test Framework - Ztest
+#############################
+
+The Zephyr Test Framework (Ztest) provides a simple testing framework intended to be used during development.
+It provides basic assertion macros and a generic test structure. The framework can be used in two ways,
+either as a generic framework for integration testing, or for unit testing specific modules.
+
+Each test name has to be unique, prefixed with test\_ and documented using doxygen.
+
+Debug and result analysis using prints are done using TC_PRINT function.
+
+Detailed explanation about the APIs and its usage in test application is given in this document.
+http://docs.zephyrproject.org/subsystems/test/ztest.html
+
+Test Methodology
+################
+
+**Test Setup**
+
+Zephyr verification setup contains
+
+#. Test Client
+
+   Test client triggers the test, the test can be triggered component wise or test suite wise.
+
+#. Device Under Test
+
+   Device Under Test could be various boards belonging to different architectures supported by Zephyr.
+
+#. Test Server or CI
+
+   Test server or CI infrastructure executes the test triggered by the client which has capabilities to
+   efficiently manage load and resource constraints.
+
+#. Test report generation tool
+
+   Test report generation tool extracts the logs generated by the execution process and provides a user friendly
+   report of test execution pass or fail with information to root cause the failures.
+
+**Test Execution**
+
+Zephyr tests can be executed in two ways:
+
+#. Manual execution
+
+   Tests are run manually on the test client and Device Under Test is connected to the Test client itself.
+   Reports are manually feed to report generation tool to obtain the results.
+
+#. Automated execution
+
+   Tests are run on test servers or CI, where tests are triggered through test client and entire test execution
```

Comment:
I'd break this into two sentences:

    Tests are run on test servers or CI, where tests are triggered
    through the test client.  The entire test execution and report
    generation is taken care automatically.